.github/workflows/rust.yml

```diff
@@ -1,28 +1,33 @@
-name: Cargo Build & Test
+name: Cargo & Clippy Test
 
 on:
   push:
+    branches: [master]
   pull_request:
+  workflow_dispatch:
 
 env:
   CARGO_TERM_COLOR: always
 
 jobs:
   build_and_test:
-    name: Rust build and tests
+    name: Rust and clippy tests
     runs-on: ubuntu-latest
-
     steps:
-    - uses: actions/checkout@v4
-    - name: Build
-      run: cargo build --verbose
-    - name: Cache dependencies installed with cargo
-      uses: actions/cache@v3
-      with:
-        path: |
-            ./target/deps
+      - uses: actions/checkout@v4
+      - uses: dtolnay/rust-toolchain@stable
+      - name: Cache dependencies installed with cargo
+        uses: actions/cache@v4
+        with:
+          path: |
+            ./target
```

Comment:
My only doubt is caching target, I changed that a week ago because it was caching also the amber build and we don't want that.
We have tests that will execute Amber manually and this creates issues.

Reply:
I think its no matter to cache amber build, but if you don't want that, i will restore it.